src/qibocal/protocols/flux_dependence/utils.py

```diff
@@ -30,6 +30,19 @@ def bias_range(self) -> np.ndarray:
         return np.arange(-self.bias_width / 2, self.bias_width / 2, self.bias_step)
 
 
+def normalize(
+    x: list[float], estimate: list[float], threshold: float = 0.001
+) -> list[float]:
+    """Normalize input array to 1.
+
+    Estimate is a guess for the mean. If the normalization
+    is off by a factor larger than 5% estimate is ignored."""
+    mean_ = np.mean(x[len(x) // 2 :])
+    if np.array(np.abs(mean_ / estimate)).max() > threshold:
+        return (np.array(x) / mean_).tolist()
+    return np.abs(np.array(x) / estimate).tolist()
```

Comment:
I don't get why guessing the mean and why computing the mean to just half of the points.

Also I think the condition in the if statement is not correct, maybe better do this way:

```suggestion
    Estimate is a guess for the mean. If the normalization
    is off by a factor larger than 5% estimate is ignored."""
    mean_ = np.mean(x[len(x) // 2 :])
    if np.array(np.abs((mean_  - estimate) / mean_)).max() > threshold:
        return (np.array(x) / mean_).tolist()
    return np.abs(np.array(x) / estimate).tolist()
```
?

Also in docstring they say the threshold is 5%, but then the default value for `threshold` is `0.001`.

Also both `x`and `estimate` are defined as 1D list of floating numbers, but then `np.mean(x)` should always return an array (hence in the following line `.max()` is useless); also, since `estimate` should be an estimation for the signal mean, why is it a list? shouldn't it be a scalar?


maybe we can simply rewrite the whole function as:

```python
def normalize(
    x: list[float]
) -> list[float]:
    """Normalize input array to 1."""
    mean_ = np.mean(x[len(x)//2 :])
   return (np.array(x) / mean_).tolist()
```